app/components/top-nav-bar.jsx

```diff
@@ -48,15 +49,46 @@ const TopNavBar = props => {
   // taken from button.jsx
   const handleKeyDown = (e, type, index) => {
     e.stopPropagation()
+
     // if space key is pressed
     if (e.keyCode === 32) {
       if (type === 'menu') {
-        handleMouseLeave() // closes any dropdowns still open
+        if (mode === 'vertical') {
+          handleMenuItemClick(menu[index])
+        } else {
+          handleMouseLeave() // closes any dropdowns still open
+        }
       } else if (type === 'menuGroup') {
         handleMouseEnter(index)
       }
     }
   }
+
+  if (mode === 'vertical') {
+    return (
+      <menu className={cx(classes.verticalMenuContainer)}>
```

Comment:
Need to apply className and ...otherprops here so the parent can pass something in to set the width.

Also, please create a test case for the story passing in a 20rem or so width.